src/actions.ts

```diff
@@ -0,0 +1,42 @@
+import * as core from '@actions/core'
+import * as spin from './spin'
+import * as github from './github'
+import * as cloud from './cloud'
+
+export async function setup(): Promise<void> {
+    let version = core.getInput('version');
+    if (!version || version === 'latest') {
+        version = await github.getLatestRelease()
+    }
+
+    await spin.install(version)
+
+    //todo: check compatibility with spin version
+    const pluginsList = core.getInput('plugins') !== '' ? core.getInput('plugins').split(',') : [];
+    if (pluginsList.length > 0) {
+        await spin.installPlugins(pluginsList)
```

Comment:
nit: at some point it might be good to rationalise whether to use semicolons or not. That can be deferred to a subsequent linting PR though.